### PR TITLE
Added doxigen documentation link

### PR DESCRIPTION
**Pull request by gcomoretto**

### CODE REVIEW

File: releases/note-source/v17_0.rst [hunk 1]
```diff
@@ -21,6 +21,7 @@ Spring 2019 Release (v17_0)
 - :doc:`Installation instructions <../install/index>`
 - :doc:`Known issues </known-issues>`
 - `Characterization Metric Report (DMTR-??) <https://ls.st/DMTR-??>`_
+- `Doxigen Documentation <http://doxygen.lsst.codes/stack/doxygen/xlink_master_2019_02_15_09.16.46/>`_
```
Comment (jdswinbank) [inline releases/note-source/v17_0.rst]: Typo: DoxYgen.

Comment (gcomoretto) [reply releases/note-source/v17_0.rst]: Typo corrected.

**Comment by jonathansick** [comment]
```
- `Characterization Metric Report (DMTR-??) <https://ls.st/DMTR-??>`_
```

@gcomoretto I just noticed this; I guess it went in with another PR, but I'm wondering if we can change the release procedures so that we never merge a PR that contains unfinished content like this. I recall from the last release that this led to an error, and I can see it kind of happening again. Once one merges a PR it's easy to forget that there's unfinished work.

**Comment by gcomoretto** [comment]
@jonathansick, this PR is going in the 17.0.x branch and not into master. There is an issue opened for the characterization report (DM-17850) that is related to the release issue.

the 17.0.x branch, in any case, will require an additional PR before merging to master, to catch the problem in case this is forgotten. But I don't think it will.

**Comment by timj** [comment]
We should get the document handle from docushare. There is no reason to have uncertainty here.

**Comment by jonathansick** [comment]
> the 17.0.x branch, in any case, will require an additional PR before merging to master, to catch the problem in case this is forgotten. But I don't think it will.

@gcomoretto about that too, https://sqr-016.lsst.io doesn't talk about what's with the release branch but I'd like to propose that a safer route is:

1. Maintain 17.0.x as a long-lived release branch
2. Cherry pick the relevant commits from this 17.0.x branch to a *ticket branch* based off the current `master` that will be merged into `master`.

I like this because it seems like a great way to make a clean merge to `master` (no need to rebase this release branch) while keeping a release branch that can be used for typo fixes for the homepage during the life of the 17.0.x releases (again, because we're not merging the `17.0.x` branch to `master`). What do you think of that?

**Comment by gcomoretto** [comment]
In the scope of this PR, I would like to close the activities on DM-996.

Regarding what to do with the 17.0.x release branch, I would like to keep the same procedure used for the 16.0 release.

Afterward, we can set up a discussion and see what the best way to improve the documentation aspects of the release.

**Comment by jonathansick** [comment]
For sure. The PR is already approved. I'm just bringing up these points since there doesn't seem to be a centralized place to talk about concerns with the release process other than in individual tickets (feel free to point me to a Slack channel otherwise!)

**Comment by gcomoretto** [comment]
OK. I will move forward with the release branch.

There are a couple of places where we can discuss the change of the release documentation: on a specific RFC (?), on Jira issue DM-14500 or on DMTN-106 that should be almost ready for discussion.

**Pull request merged**